tests/cpp/test_multidevice_pipeline.cpp

```diff
@@ -147,6 +177,7 @@ TEST_P(PipelineTestTwoStages, Communication) {
        is_stage0_sharded,
        is_stage1_sharded,
        do_reduction,
+       sharded_dim,
        use_fusion_executor_cache] = GetParam();
   if (!disable_skip && !communicator->isBackendAvailable(backend)) {
```

Comment:
What's `disable_skip` for? Is it a duplicate of the existing `--gtest_also_run_disabled_tests`?

Reply:
I'm asking this because tests should be skipped for good reasons, e.g., the test requires 6 GPUs but only 2 found. I don't understand why one would attempt to disable the skipping. If a test currently fails because of a bug, the test should be marked as `DISABLED_` and `--gtest_also_run_disabled_tests` is the right way to enforce to run them.

Reply:
It was a flag used for debugging, so that we could run tests with 1 device and only see 1 processes' print statements.
I didn't know about this flag in gtest, so we can use that instead and remove the disable_skip! FYI @samnordmann just be aware of this update

Reply:
> we could run tests with 1 device and only see 1 processes' print statements.

Interesting. In that case, does that process run the program on sharded tensors? If yes, how does stuff like allgather run? If no, should we make one-device a test that's always enabled?

Reply:
The program will not run correctly! 
This is helpful for debugging the test when the bug is in compilation before it executes.

Reply:
> If no, should we make one-device a test that's always enabled?
I think we could benefit from some more one-device only tests that check the behavior of compilation without execution. We have a couple, but as a TODO item we should try and improve the one-device test coverage where possible.

Reply:
This flag is a debug utility for when we need to debug something happening during lowering (which is often the case), and to allow us to use gdb (because debuggin distributed program is very annoying). So, no network communication will be executed, the program will crash at some point, but it's still a good debugging tool that I'm using on a regular basis